### PR TITLE
Dvc remote default in list validation 

### PR DESCRIPTION
**Pull request by karajan1001**
* [x] ❗ I have followed the [Contributing to DVC](https://dvc.org/doc/user-guide/contributing/core) checklist.

* [x] 📖 If this PR requires [documentation](https://dvc.org/doc) updates, I have created a separate PR (or issue, at least) in [dvc.org](https://github.com/iterative/dvc.org) and linked it here. If the CLI API is changed, I have updated [tab completion scripts](https://github.com/iterative/dvc/tree/master/scripts/completion).

* [x] ❌ I will check DeepSource, CodeClimate, and other sanity checks below. (We consider them recommendatory and don't expect everything to be addressed. Please fix things that actually improve code or fix bugs.)

Thank you for the contribution - we'll try to review it as soon as possible. 🙏

### CODE REVIEW

File: tests/func/test_config.py [hunk 1]
```diff
@@ -78,7 +78,7 @@ def test_non_existing(self):
         self.assertEqual(ret, 251)
 
         ret = main(["config", "core.remote", "myremote"])
-        self.assertEqual(ret, 0)
```
Comment (karajan1001) [inline tests/func/test_config.py:81]: All other config settings failed in this function.

File: dvc/config.py [hunk 2]
```diff
@@ -238,12 +239,11 @@ def __init__(self, dvc_dir=None, validate=True):
     def get_dir(cls, level):
         from appdirs import user_config_dir, site_config_dir
 
-        assert level in ("global", "system")
```
Comment (karajan1001) [inline dvc/config.py:241]: These changes are required by deepsource

Comment (efiop) [reply dvc/config.py:241]: @karajan1001 DS is a bit strict on asserts in code, we just ignore those complaints.

**Comment by efiop** [comment]
Sorry for the delay! Thank you so much @karajan1001 !

**Pull request merged**